packages/mobx-state-tree/__tests__/core/async.test.ts

```diff
@@ -356,3 +357,51 @@ test("flow typings", async () => {
         expect(_d).toBe(2)
     })
 })
+
+/**
+ * Detect explicit `any` type.
+ * https://stackoverflow.com/a/55541672/4289902
+ */
+type IfAny<T, Y, N> = 0 extends 1 & T ? Y : N
+function ensureNotAny<T>(value: IfAny<T, never, T>) {}
+function ensureType<T>(value: T) {}
+
+test("yield* typings", async () => {
+    const voidPromise = () => Promise.resolve()
+    const numberPromise = () => Promise.resolve(7)
+    const stringWithArgsPromise = (input1: string, input2: boolean) =>
+        Promise.resolve("test-result")
+
+    const voidGen = toGenerator(voidPromise)
+    const numberGen = toGenerator(numberPromise)
+    const stringWithArgsGen = toGenerator(stringWithArgsPromise)
+
+    const M = types.model({ x: 5 }).actions((self) => {
+        function* testAction() {
+            const voidResult = yield* voidGen()
+            ensureNotAny(voidResult)
+            ensureType<void>(voidResult)
+
+            const numberResult = yield* numberGen()
+            ensureNotAny(numberResult)
+            ensureType<number>(numberResult)
+
+            const stringResult = yield* stringWithArgsGen("input", true)
+            ensureNotAny(stringResult)
+            ensureType<string>(stringResult)
+
+            return stringResult
+        }
+
+        return {
+            testAction: flow(testAction)
+        }
+    })
+
+    const m = M.create()
+
+    const result = await m.testAction()
+    ensureNotAny(result)
```

Comment:
Now that the repo is upgraded to ts 3.9, we may use 
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments
for this kind of assertions

Reply:
I don't think I can use `@ts-expect-error` in this case?  If I understand correctly, it's for when typescript wants to report an error, but in the context of the test it's the intended code.  

In this test I'm wanting the opposite, as what I'm testing for is that the type of the returned value is "not `any`", which never normally shows up as an error of course.  With the current code, if I change one of the `yield*` calls to just be `yield`, a typescript error appears (as intended).  

I've simplified the check from two functions to one, and added a comment.  Let me know if I've misunderstood though!

Reply:
What you cloud do: 
```ts
// result return type is string. if there is no type error here, means we have wrong return type
// @ts-expect-error
const notAny: number =  result
;
const isString: string =  result;
```

Reply:
I didn't think of that approach - nice!

I couldn't think of a way to have a "one-liner" to do both checks though (a la the `ensureNotAnyType` function).  Would you prefer to keep the tests as they are now, or switch them to the technique above?